### PR TITLE
[GraphBolt] Add more detail to CopyTo docstring

### PR DESCRIPTION
**Pull request by RamonZhou**
## Description
Clarify the attribute selection in CopyTo.

## Checklist
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [ ] I've leverage the [tools](https://docs.google.com/document/d/1iHyj7zlmygKSk5gBPsqIqL5ASPzJSPREaNT_QdsiYA4/edit) to beautify the python and c++ code.
- [ ] The PR is complete and small, read the [Google eng practice (CL equals to PR)](https://google.github.io/eng-practices/review/developer/small-cls.html) to understand more about small PR. In DGL, we consider PRs with less than 200 lines of core code change are small (example, test and documentation could be exempted).
- [ ] All changes have test coverage
- [ ] Code is well-documented
- [ ] To the best of my knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR
- [ ] If the PR is for a new model/paper, I've updated the example index [here](../examples/README.md).

### CODE REVIEW

File: python/dgl/graphbolt/base.py [hunk 1]
```diff
@@ -108,22 +108,38 @@ class CopyTo(IterDataPipe):
...
 
-    This is equivalent to
+    When `data` has `to` method implemented, `CopyTo` will be equivalent to
 
     .. code:: python
 
        for data in datapipe:
            yield data.to(device)
 
+    For `gb.MiniBatch`, only a part of attributes will be transferred to
```
Comment (Rhett-Ying) [inline python/dgl/graphbolt/base.py:118]: `:class:~dgl.graphbolt.MiniBatch`

File: python/dgl/graphbolt/base.py [hunk 2]
```diff
@@ -108,22 +108,38 @@ class CopyTo(IterDataPipe):
...
+    accelerate the process:
```
Comment (Rhett-Ying) [inline python/dgl/graphbolt/base.py:119]:
```suggestion
    accelerate the process in default:
```

File: python/dgl/graphbolt/base.py (hunk 1 above)
Comment (Rhett-Ying) [reply python/dgl/graphbolt/base.py:118]: refer to https://github.com/dmlc/dgl/blob/557a8f817a285acc9dfee91c19a19ff8728a8d76/python/dgl/dataloading/dataloader.py#L789

File: python/dgl/graphbolt/base.py [hunk 3]
```diff
@@ -108,22 +108,38 @@ class CopyTo(IterDataPipe):
...
+
+    - When `seed_nodes` is not None and `node_pairs` is None, node related
```
Comment (Rhett-Ying) [inline python/dgl/graphbolt/base.py:121]: usually we use double backticks such s "``seed_nodes``"

File: python/dgl/graphbolt/base.py [hunk 4]
```diff
@@ -108,22 +108,38 @@ class CopyTo(IterDataPipe):
...
+    task is inferred. Only `labels`, `sampled_subgraphs`, `node_features`
+    and `edge_features` will be transferred.
+
+    - When `node_pairs` is not None and `seed_nodes` is None, edge/link
+    related task is inferred. Only `labels`, `compacted_node_pairs`,
+    `compacted_negative_srcs`, `compacted_negative_dsts`, `sampled_subgraphs`,
+    `node_features` and `edge_features` will be transferred.
+
+    - Otherwise, all attributes will be transferred. If you want some other
+    attributes to be transferred as well, please indicate the name in the
```
Comment (Rhett-Ying) [inline python/dgl/graphbolt/base.py:131]: put `If you xxx` into another separate line as `extra_attrs` is the way to copy additional atrributes to target device no matter what kind of tasks are inferred, even not limited to `MiniBatch` only, it could be applied to other data objects too.

File: python/dgl/graphbolt/base.py [hunk 5]
```diff
@@ -108,22 +108,38 @@ class CopyTo(IterDataPipe):
...
+    `extra_attrs`.
```
Comment (Rhett-Ying) [inline python/dgl/graphbolt/base.py:132]: add an detailed example to showcase how to copy extra attributes such as `input_nodes` via `extra_attributes`.

File: python/dgl/graphbolt/base.py [hunk 6]
```diff
@@ -108,22 +108,38 @@ class CopyTo(IterDataPipe):
...
     Parameters
     ----------
     datapipe : DataPipe
         The DataPipe.
     device : torch.device
         The PyTorch CUDA device.
     extra_attrs: List[string]
-        The extra attributes in the MiniBatch you want to be carried to the
-        specific device.
+        The extra attributes of the data in the DataPipe you want to be carried
```
Comment (Rhett-Ying) [inline python/dgl/graphbolt/base.py:141]: pls remember to add dedicated testcases for `extra_attrs` in this PR or another PR.

Comment (RamonZhou) [reply python/dgl/graphbolt/base.py:141]: Okay

File: python/dgl/graphbolt/base.py [hunk 7]
```diff
@@ -108,22 +108,46 @@ class CopyTo(IterDataPipe):
...
 
-    This is equivalent to
+    When `data` has `to` method implemented, `CopyTo` will be equivalent to
 
     .. code:: python
 
        for data in datapipe:
            yield data.to(device)
 
+    For :class:~dgl.graphbolt.MiniBatch, only a part of attributes will be
```
Comment (Rhett-Ying) [inline python/dgl/graphbolt/base.py:118]:
```suggestion
    For :class:`~dgl.graphbolt.MiniBatch`, only a part of attributes will be
```

File: python/dgl/graphbolt/base.py [hunk 8]
```diff
@@ -108,22 +108,46 @@ class CopyTo(IterDataPipe):
...
+    transferred to accelerate the process by default:
+
+    - When "seed_nodes" is not None and "node_pairs" is None, node related
```
Comment (Rhett-Ying) [inline python/dgl/graphbolt/base.py:121]:
```suggestion
    - When ``seed_nodes`` is not None and ``node_pairs`` is None, node related
```

File: python/dgl/graphbolt/base.py [hunk 9]
```diff
@@ -108,22 +108,46 @@ class CopyTo(IterDataPipe):
...
+    task is inferred. Only "labels", "sampled_subgraphs", "node_features"
```
Comment (Rhett-Ying) [inline python/dgl/graphbolt/base.py:122]: please use ```` instead of """

File: python/dgl/graphbolt/base.py [hunk 10]
```diff
@@ -108,22 +108,46 @@ class CopyTo(IterDataPipe):
...
+
+    - When "node_pairs" is not None and "seed_nodes" is None, edge/link
+    related task is inferred. Only "labels", "compacted_node_pairs",
+    "compacted_negative_srcs", "compacted_negative_dsts", "sampled_subgraphs",
+    "node_features" and "edge_features" will be transferred.
+
+    - Otherwise, all attributes will be transferred.
+
+    - If you want some other attributes to be transferred as well, please
+    indicate the name in the `extra_attrs`. For instance, the following code
```
Comment (Rhett-Ying) [inline python/dgl/graphbolt/base.py:133]: indicate -> specify

File: python/dgl/graphbolt/base.py [hunk 11]
```diff
@@ -108,22 +108,46 @@ class CopyTo(IterDataPipe):
...
         The DataPipe.
     device : torch.device
         The PyTorch CUDA device.
     extra_attrs: List[string]
-        The extra attributes in the MiniBatch you want to be carried to the
-        specific device.
+        The extra attributes of the data in the DataPipe you want to be carried
+        to the specific device. The attributes indicated in the `extra_attrs`
+        will be transferred regardless of the task inferred. It could also be
+        applied to classes other than :class:~dgl.graphbolt.MiniBatch.
```
Comment (Rhett-Ying) [inline python/dgl/graphbolt/base.py:150]:
```suggestion
        applied to classes other than :class:`~dgl.graphbolt.MiniBatch`.
```

File: python/dgl/graphbolt/base.py [hunk 12]
```diff
@@ -108,22 +108,46 @@ class CopyTo(IterDataPipe):
...
     ----------
     datapipe : DataPipe
         The DataPipe.
     device : torch.device
         The PyTorch CUDA device.
     extra_attrs: List[string]
-        The extra attributes in the MiniBatch you want to be carried to the
-        specific device.
+        The extra attributes of the data in the DataPipe you want to be carried
+        to the specific device. The attributes indicated in the `extra_attrs`
```
Comment (Rhett-Ying) [inline python/dgl/graphbolt/base.py:148]:
```suggestion
        to the specific device. The attributes indicated in the ``extra_attrs``
```

**Review by Rhett-Ying — APPROVED** [review]
